src/main/java/Duke.java

```diff
@@ -1,10 +1,153 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+    private static String logo  = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static ArrayList<Task> list;
+
+    public static void main(String[] args) throws DukeException {
+        list = new ArrayList<>();
+
         System.out.println("Hello from\n" + logo);
+        System.out.println("What can I do for you today, Master?");
+        Scanner scanner = new Scanner(System.in);
+        String text = scanner.nextLine().toLowerCase();
+        boolean open = true;
+        while (open) {
+            try {
+                // Bye
+                if (text.equals("bye") || text.equals("exit")) {
+                    scanner.close();
+                    open = false;
+                    System.out.println("Goodbye, Master! Thank you for visiting\n" + logo);
+                    // List
+                } else if (text.equalsIgnoreCase("list")) {
+                    if (list.isEmpty()) {
+                        throw new DukeException("There is nothing in your list yet!");
+                    } else {
+                        System.out.println("Here is your to-do list, Master:");
```

Comment:
Would this be better done within TaskList, or in a Command class? Also, one other recommendation would be for outputs to be displayed via a Ui class :)